File: arena-junit/src/main/java/arena/junit/lifecycle/ComponentState.java
package arena.junit.lifecycle;

import java.util.List;

public final class ComponentState {
  public String id = "";
  public String state = "";
  public List<Fault> faults = List.of();
  public List<ComponentState> children = List.of();

  public ComponentState find(String identifier) {
    if (id.equals(identifier)) {
      return this;
    }
    for (ComponentState child : children) {
      ComponentState found = child.find(identifier);
      if (found != null) {
        return found;
      }
    }
    return null;
  }
}

File: arena-xunit/src/Lifecycle/ComponentState.cs
using System.Collections.Generic;

namespace ArenaDotnet.Xunit.Lifecycle;

public sealed class ComponentState
{
    public string Id { get; set; } = "";
    public string State { get; set; } = "";
    public List<Fault> Faults { get; set; } = new();
    public List<ComponentState> Children { get; set; } = new();

    public ComponentState? Find(string identifier)
    {
        if (Id == identifier)
            return this;
        foreach (var child in Children)
        {
            var found = child.Find(identifier);
            if (found != null)
                return found;
        }
        return null;
    }
}
